Druid/DRReceiverConstraint.class.st

```diff
@@ -0,0 +1,32 @@
+Class {
+	#name : #DRReceiverConstraint,
+	#superclass : #Object,
+	#instVars : [
+		'constraint'
+	],
+	#category : #'Druid-IR-Paths'
+}
+
+{ #category : #'instance creation' }
+DRReceiverConstraint class >> with: aConstraint [
+
+	^ self new
+		  constraint: aConstraint;
+		  yourself
+]
+
+{ #category : #adding }
+DRReceiverConstraint >> addToEdge: aDRCPSEdge [
+
+	| loadReceiverInstructions |
+	loadReceiverInstructions := aDRCPSEdge source controlFlowGraph
+		                            instructions select: [ :e |
+		                            e isLoadReceiver ].
+	loadReceiverInstructions do: [ :instruction |
+		aDRCPSEdge addConstraint: constraint to: instruction result name ]
+]
```

Comment:
What do you want to do here? This doesn't seem good because the edge has info about the paths before, and here you are taking the info from all the CFG.

Reply:
Ok, what we cannot say now is the register where is the receiver, right?

In that case, maybe we need to check what the visitor is doing with the DRLoadReceiver instruction (maybe now is doing nothing).
The problem is how to define the specialization in that case, we need to think about that.

Reply:
Anyways this does not seem to have bad side effects for now, as the LoadReceiver instruction is always at the top of the cfg